src/sentry/tasks/deletion.py

```diff
@@ -129,7 +139,14 @@ def delete_project(object_id, transaction_id=None, continuous=True, **kwargs):
         p.update(status=ProjectStatus.DELETION_IN_PROGRESS)
 
     # Immediately revoke keys
-    ProjectKey.objects.filter(project_id=object_id).delete()
+    project_keys = ProjectKey.objects.filter(project_id=object_id)
+    deleted_keys = [pk.id for pk in project_keys]
+    project_keys.delete()
```

Comment:
Note though, that this is still subject to a race condition, so it might not be worth logging imo.